### ISSUE TITLE
Fixed the None grad of the encoder on GPU

### ISSUE THREAD

**Issue by KyoungHwanLee** [CorentinJ/Real-Time-Voice-Cloning#237]
Hello. Thank you for opening the great project to everyone.

You commented
`# FIXME: currently, the gradient is None if loss_device is cuda`
on 32 line of encoder/train.py, and I think I found a solution.

After some research, I found that copying a tensor with requires_grad==True to another device occurs 'gradient lost'. ([https://towardsdatascience.com/understanding-pytorch-with-an-example-a-step-by-step-tutorial-81fc5f8c4e8e](https://towardsdatascience.com/understanding-pytorch-with-an-example-a-step-by-step-tutorial-81fc5f8c4e8e), 'Creating Parameters' Section)

So, modifying the 27-28 lines of encoder/model.py
` self.similarity_weight = nn.Parameter(torch.tensor([10.])).to(loss_device)
  self.similarity_bias = nn.Parameter(torch.tensor([-5.])).to(loss_device)`
to

` self.similarity_weight = nn.Parameter(torch.tensor([10.]).to(loss_device))
  self.similarity_bias = nn.Parameter(torch.tensor([-5.]).to(loss_device))`
or
` self.similarity_weight = nn.Parameter(torch.tensor([10.], device=loss_device))
  self.similarity_bias = nn.Parameter(torch.tensor([-5.], device=loss_device))`

and set the 'loss_device' to cuda fixed the None grad problem on GPU with improving the training speed.
In my case - batch size 64 with V100 GPU :
loss on CPU : 16.4 mins for 1k steps 
loss on GPU :  13.4 mins for 1k steps
